Errorring test under NFS: Empty directories cannot be Zarr assets

few days ago testing under NFS was added .  In https://github.com/dandi/dandi-cli/pull/1267#issuecomment-1497873494 a run under NFS failed with exception
```
_________________ test_dandi_file_zarr_with_excluded_dotfiles __________________
dandi/tests/test_files.py:309: in test_dandi_file_zarr_with_excluded_dotfiles
    zf = dandi_file(zarr_path)
dandi/files/__init__.py:195: in dandi_file
    return factory(filepath, path, dandiset_path)
dandi/files/_private.py:71: in __call__
    return self.CLASSES[DandiFileType.classify(filepath)](
dandi/files/_private.py:42: in classify
    raise UnknownAssetError("Empty directories cannot be Zarr assets")
E   dandi.exceptions.UnknownAssetError: Empty directories cannot be Zarr assets
```
which did not emerge upon rerun.

But apparently it was not the first time it failed - `master` experienced them already

```
(dandisets-2) dandi@drogon:/mnt/backup/dandi/tinuous-logs/dandi-cli/2023$ datalad foreach-dataset --output-streams relpath git grep 'Empty directories cannot be Zarr assets'
04/04/github/push/master/0be71d3/Tests/4856/18_test (ubuntu-latest, 3.7, nfs).txt:2023-04-04T16:14:55.1866665Z     raise UnknownAssetError("Empty directories cannot be Zarr assets")
04/04/github/push/master/0be71d3/Tests/4856/18_test (ubuntu-latest, 3.7, nfs).txt:2023-04-04T16:14:55.1867133Z E   dandi.exceptions.UnknownAssetError: Empty directories cannot be Zarr assets
04/04/github/push/master/0be71d3/Tests/4856/18_test (ubuntu-latest, 3.7, nfs).txt:2023-04-04T16:14:55.1877404Z FAILED dandi/tests/test_files.py::test_dandi_file_zarr_with_excluded_dotfiles - dandi.exceptions.UnknownAssetError: Empty directories cannot be Zarr assets
04/04/github/push/master/0be71d3/Tests/4856/test (ubuntu-latest, 3.7, nfs)/7_Run all tests.txt:2023-04-04T16:14:55.1866660Z     raise UnknownAssetError("Empty directories cannot be Zarr assets")
04/04/github/push/master/0be71d3/Tests/4856/test (ubuntu-latest, 3.7, nfs)/7_Run all tests.txt:2023-04-04T16:14:55.1867128Z E   dandi.exceptions.UnknownAssetError: Empty directories cannot be Zarr assets
04/04/github/push/master/0be71d3/Tests/4856/test (ubuntu-latest, 3.7, nfs)/7_Run all tests.txt:2023-04-04T16:14:55.1877393Z FAILED dandi/tests/test_files.py::test_dandi_file_zarr_with_excluded_dotfiles - dandi.exceptions.UnknownAssetError: Empty directories cannot be Zarr assets
```

so added testing scenario already picking up on some NFS oddity.  Should be looked into what is happening -- most likely that NFS is not fast enough to show the files listing for the freshly generated test zarr folder?  then we might need to add some sleeping but such situation sounds a bit odd -- needs analysis.

--- a/dandi/tests/test_files.py
+++ b/dandi/tests/test_files.py
@@ -1,4 +1,5 @@
 from operator import attrgetter
+import os
 from pathlib import Path
 import subprocess
 from typing import cast
@@ -305,7 +306,11 @@ def test_dandi_file_zarr_with_excluded_dotfiles(tmp_path: Path) -> None:
     )
     with pytest.raises(UnknownAssetError):
         dandi_file(zarr_path)
-    (zarr_path / "arr_0" / "foo").touch()
+    with (zarr_path / "arr_0" / "foo").open("w") as fp:
+        print("Text.", file=fp)
+        # Force changes to be synced when testing on NFS:
+        fp.flush()
+        os.fsync(fp.fileno())
     zf = dandi_file(zarr_path)
     assert isinstance(zf, ZarrAsset)
 

